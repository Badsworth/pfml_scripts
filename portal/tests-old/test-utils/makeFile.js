/**
 * Make a File instance
 * @see https://developer.mozilla.org/en-US/docs/Web/API/File
 * @param {object} attrs File attributes
 * @param {string} attrs.name The file's name
 * @param {string} attrs.type The file's MIME type
 * @returns {File}
 */
export const makeFile = (attrs = {}) => {
  const { name, type } = Object.assign(
    {
      name: "file.pdf",
      type: "application/pdf",
    },
    attrs
  );

  return new File([], name, { type });
};

/**
 * Create an HTML <input> for a test's event.target
 * @param {object} attrs - HTML attributes for the input
 * @returns {HTMLInputElement}
 */
export function createInputElement(attrs) {
  const input = document.createElement("input");

  Object.entries(attrs).forEach(([attributeName, value]) => {
    input.setAttribute(attributeName, value);
  });

  if (attrs.checked === false) input.checked = false;

  return input;
}
