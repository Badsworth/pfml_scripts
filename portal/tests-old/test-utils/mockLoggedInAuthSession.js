import { Auth } from "@aws-amplify/auth";

/**
 * Mock the Cognito authentication session so a test user appears logged in
 */
const mockLoggedInAuthSession = () => {
  const accessTokenJwt =
    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiQnVkIn0.YDRecdsqG_plEwM0H8rK7t2z0R3XRNESJB5ZXk-FRN8";

  jest.spyOn(Auth, "currentSession").mockImplementation(() =>
    Promise.resolve({
      accessToken: { jwtToken: accessTokenJwt },
    })
  );
};

export default mockLoggedInAuthSession;
