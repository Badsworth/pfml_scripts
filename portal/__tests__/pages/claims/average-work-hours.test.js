import AverageWorkHours from "../../../src/pages/claims/average-work-hours";
import { renderWithAppLogic } from "../../test-utils";

jest.mock("../../../src/hooks/useAppLogic");

describe("AverageWorkHours", () => {
  let appLogic, wrapper;

  beforeEach(() => {
    ({ appLogic, wrapper } = renderWithAppLogic(AverageWorkHours, {
      claimAttrs: {
        temp: { leave_details: { avg_weekly_work_hours: "40" } },
      },
    }));
  });

  it("renders the form", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("when the form is successfully submitted", () => {
    it("calls updateClaim", () => {
      wrapper.find("QuestionPage").simulate("save");

      expect(appLogic.updateClaim).toHaveBeenCalledTimes(1);
    });
  });
});
