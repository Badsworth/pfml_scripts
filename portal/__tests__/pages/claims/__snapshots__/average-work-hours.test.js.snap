// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AverageWorkHours renders the form 1`] = `
<QuestionPage
  onSave={[Function]}
  title="Leave details"
>
  <InputText
    hint="We need to know this so we can calculate your benefit amount. If your hours vary from week to week, tell us the average over the past 12 months."
    label="How many hours do you work for your employer, on average?"
    name="temp.leave_details.avg_weekly_hours_worked"
    onChange={[Function]}
    value="40"
    width="small"
  />
</QuestionPage>
`;
